pkg/querier/querier.go

```diff
@@ -151,17 +145,15 @@ func (q *chunkQuerier) Select(sortSeries bool, hints *storage.SelectHints, match
 	return storage.NewSeriesSetToChunkSet(q.Querier.Select(sortSeries, hints, matchers...))
 }
 
-// QueryableWithFilter extends Queryable interface with `UseQueryable` filtering function.
-type QueryableWithFilter interface {
-	storage.Queryable
-
-	// UseQueryable returns true if this queryable should be used to satisfy the query for given time range.
-	// Query min and max time are in milliseconds since epoch.
-	UseQueryable(now time.Time, queryMinT, queryMaxT int64) bool
+// Queryable interface is similar to storage.Queryable, but it can return nil querier, if query should not use this queryable.
```

Comment:
Can you add some extra information that would help people using this interface like:
* `Querier` will be nil if this `Queryable` shouldn't be used for a particular query. Will error be `nil` in this case?
* Is it OK for some implementations to always return a `Querier`? Never return a `Querier`?